lib/client/api.go

```diff
@@ -1918,12 +1918,20 @@ func (tc *TeleportClient) applyProxySettings(proxySettings ProxySettings) error
 			tc.KubeProxyAddr = proxySettings.Kube.PublicAddr
 		// ListenAddr is the second preference.
 		case proxySettings.Kube.ListenAddr != "":
-			if _, err := utils.ParseAddr(proxySettings.Kube.ListenAddr); err != nil {
+			addr, err := utils.ParseAddr(proxySettings.Kube.ListenAddr)
+			if err != nil {
 				return trace.BadParameter(
 					"failed to parse value received from the server: %q, contact your administrator for help",
 					proxySettings.Kube.ListenAddr)
 			}
-			tc.KubeProxyAddr = proxySettings.Kube.ListenAddr
```

Comment:
Nit: Technically, you could've just kept this line and not include "else" branch below, but I guess this is a matter of preference (an explicit "else").

Reply:
Yeah, this is just to be explicit